1257/en.md

```diff
@@ -0,0 +1,447 @@
+![table3](table3.png) <br>
+Now what will be the maximum distance from node `5`? Distance between node `0` and node `5` is ```1+2+3+1+2 = 9``` and distance between node `5` and node `9` is ```1+2+3+2 = 8```. If we observe, we can see that, maximum distance for each node will be either distance from node `0` or distance from node `9`. <br>
+
+Similarly, in a graph, maximum distance for each node is either distance from node `P` or distance from node `Q` where distance between `P` and `Q` is the maximum possible distance in the graph. 
+
+### Solution in C++: <br>
+``` cpp
+#include<bits/stdc++.h>
+
+//defines
+#define      loop(m,n)          for(m=0;m<n;m++)
+#define      loop1(m,n)         for(m=1;m<=n;m++)
+#define      rloop(m,n)         for(m=n-1;m>=0;m--)
+#define      case()             "Case " << z++ << ":"
+#define      test_inp()         ll Tc,z=1;cin>>Tc;while(Tc--)
+#define      ntest_inp()        for(int Tc=0;Tc<1;Tc++)
+#define      PTF(b)             puts(b?"YES":"NO")
+#define      Ptf(b)             puts(b?"Yes":"No")
+#define      ptf(b)             puts(b?"yes":"no")
+#define      newline            cout<<endl
+#define      quit               return 0
+
+using namespace std;
+
+//typedefs
+typedef long long ll;
+typedef pair<ll,ll> pll;
+
+//constants
+const ll mod=1000000007;
+
+//Solve here...
+vector<pll> adj[30005];
+ll vis[30005] = {0} ;
+ll dif[30005];
+ll mx = 0, mxnode ;
+
+void dfs(ll s, ll d)   //modified dfs function
+{
+    vis[s] = 1 ;
+    dif[s] = max(d, dif[s]) ;
+    if(d>mx)
+    {
+        mx = d ;
+        mxnode = s ;
+    }
+
+    ll i;
+    loop(i, adj[s].size())
+    {
+        if(vis[adj[s][i].first] == 0)
+        {
+            dfs(adj[s][i].first, d+adj[s][i].second);
+        }
+    }
+}
+
+
+int main()
+{
+    int i,j;
+
+    test_inp()
+    {
+        ll n;
+        cin >> n ;
+
+        loop(i, n-1)
+        {
+            ll a, b, w ;
+            cin >> a >> b >> w ; 
+
+            adj[a].push_back({b,w});
+            adj[b].push_back({a,w});
+        }
+
+       int P, Q ;
+        //1st step
+        mx = 0;         
+        dfs(0, 0);
+        P = mxnode ;
+
+        memset(vis, 0, sizeof vis);
+        mx = 0;
+        
+        //2nd step
+        dfs(P, 0);
+        Q = mxnode ;
+        int P_diff[n+1];
+        loop(i, n)
+            P_diff[i] = dif[i];
+
+        memset(dif, 0, sizeof dif);
+        memset(vis, 0, sizeof vis);
+        
+        //3rd step
+        dfs(Q, 0);
+        int Q_diff[n+1];
+        loop(i, n)
+            Q_diff[i] = dif[i];
+
+        //4th step
+        cout << case() << endl;
+
+        loop(i, n)
+            cout << max(P_diff[i], Q_diff[i]) << endl;
+
+        memset(adj, 0, sizeof adj);
+        memset(vis, 0, sizeof vis);
+        memset(dif, 0, sizeof dif);
+        
+    }
+
+    quit;
+}
+```
+### Solution in Python: (Thanks  to [Shanto Tanzim](https://codeforces.com/profile/Shantol) for the solution) <br>
```

Comment:
Please link their LightOJ profile instead.

Reply:
Is it possible in LightOj to view and copy other's profile link? I could'nt do that. Or should I mention their LightOJ ID name only?

Reply:
Yes it's possible. You can ask them.
However, if you're unable, I guess CF link for this one time only is alright.